### PR TITLE
Branch UG DG Additions

### PR DESCRIPTION
**Pull request by aditi2313**
Changes to UG:
1. Tutorial addition to Quick Start section.
2. Removed sample commands to test under Quick Start.

Additions to DG:
1. Implementation details and design considerations for `patron overdue` command.
2. New sequence and activity diagrams for `patron overdue` command as part of implementation details.
3. DG Manual testing for `patron overdue` command. 
4. DG Product Scope changes
5. DG Non-functional requirement changes

### CODE REVIEW

File: docs/diagrams/OverduePatronCommandSequenceDiagram.puml [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+CommandResult --> OverduePatronCommand
+deactivate CommandResult
+
+OverduePatronCommand --> LogicManager : result
+deactivate OverduePatronCommand
+
+[<--LogicManager
+deactivate LogicManager
+@enduml
```
Comment (Halpfrog) [inline docs/diagrams/OverduePatronCommandSequenceDiagram.puml:69]: LGTM!

**Review by Halpfrog — COMMENTED** [review]
LGTM

File: docs/UserGuide.md [hunk 2]
```diff
@@ -53,35 +53,140 @@ For the glossary of content, you can refer to our [Glossary](#7-glossary) sectio
...
+
+<div markdown="block" class="alert alert-info">
+
+**Notes:**<br>
+
+* The return dates of the above `borrow` commands may be changed. 
+
+* Learn more about the `borrow` command [here](#336-borrowing-a-book--borrow).
+
+</div>
```
Comment (yl-ang) [inline docs/UserGuide.md:151]: Good use of alerts to tell the user to take note of the borrow command 👍

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -975,6 +1031,39 @@ testers are expected to do more *exploratory* testing.
...
+      Expected: Patron list contains borrowers of all books borrowed.
+
+4. Displaying patrons with overdue books while there are borrowed books and some of them are overdue. 
+   1. Prerequisites: List all books using the `book list` command. Multiple books in the book list. Ensure that there 
+      are some books borrowed with return dates after the present date.
+      Ensure that there are some books borrowed with return dates before the present date.
+
+   2. Test case: `patron overdue`<br>
+      Expected: Patron list contains borrowers of all books borrowed with return dates before the present date.
+
```
Comment (yl-ang) [inline docs/DeveloperGuide.md:1066]: The testing looks good to me 💯

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -264,6 +264,57 @@ The return command is designed to be to return multiple books in one command, wh
...
+other command or additional information.
+
+LibTask can store a large number of books and patrons, making it infeasible for the user to scroll through the book list
+to identify patrons with books borrowed beyond their return dates. The design of the overdue command hence enhances 
+LibTask's usability, as librarians can experience more efficient processing of overdue books.
+
+`patron overdue` is designed to not affect the displayed book list while interacting with LibTask's patron list (for 
+example by displaying all overdue books in the displayed book list). However, executing the `book related` command in 
+succession to the overdue command gives users a more detailed view the patron's overdue books.
+
```
Comment (yl-ang) [inline docs/DeveloperGuide.md:317]: I like the addition of design consideration 💯

File: docs/UserGuide.md [hunk 5]
```diff
@@ -53,35 +53,140 @@ For the glossary of content, you can refer to our [Glossary](#7-glossary) sectio
...
+The various GUI components of LibTask are described in the picture below:
 
-   * **`patron list`** : Lists all patrons in LibTask's patron list.
+![LabeledUi](images/LabeledGUI.png)
 
-   * **`patron add n/Alice s/S01823283S p/90123212 e/profA@u.nus.edu `** : Adds a patron named `Alice` into LibTask's patron list.
+* `Menu bar`: Menu buttons that you can click to exit LibTask or to open up the help window that contains a link to this user guide.
+* `Command box`: You can type and enter commands to run in this box.
+* `Result box`: You can view the results from running your commands in this box.
+* `patron list`: You can view the list of patrons with their details in this list.
```
Comment (yl-ang) [inline docs/UserGuide.md:64]: This is an old bug corrected from PED, the newer folk repo has updated 🤔

**Review by yl-ang — CHANGES_REQUESTED** [review]
Overall LGTM! The changes are good but I think your folk didn't contain the updated the updated changes from PED prs ><

**Review by yl-ang — APPROVED** [review]
LGTM!

**Pull request merged**